### PR TITLE
cursor pagination removed and few test cases added.

### PR DESCRIPTION
**Pull request by vikasrathee-cs**
cursor pagination removed and few test cases added.

### CODE REVIEW

File: src/test/java/io/cdap/plugin/zendesk/source/batch/ZendeskBatchSourceConfigTest.java [hunk 1]
```diff
@@ -16,22 +16,34 @@
 
 package io.cdap.plugin.zendesk.source.batch;
 
+import io.cdap.cdap.api.data.format.StructuredRecord;
 import io.cdap.cdap.api.data.schema.Schema;
+import io.cdap.cdap.api.dataset.lib.KeyValue;
+import io.cdap.cdap.etl.api.Emitter;
 import io.cdap.cdap.etl.api.FailureCollector;
+import io.cdap.cdap.etl.api.batch.BatchRuntimeContext;
+import io.cdap.cdap.etl.api.batch.BatchSource;
```
Comment (sgarg-CS) [inline src/test/java/io/cdap/plugin/zendesk/source/batch/ZendeskBatchSourceConfigTest.java]: Remove unused import statements

File: src/test/java/io/cdap/plugin/zendesk/source/batch/ZendeskBatchSourceConfigTest.java [hunk 2]
```diff
@@ -322,4 +334,125 @@ public void testGetSchemaInvalidJson() {
...
+      20,
+      300,
+      300,
+      "https://%s.zendesk.com/api/v2/%s",
+      "");
+    try {
+      ZendeskBatchSource zendeskBatchSource = new ZendeskBatchSource(config);
+      pipelineConfigurer = new MockPipelineConfigurer(null);
+      zendeskBatchSource.configurePipeline(pipelineConfigurer);
+    } catch (ValidationException ve) {
```
Comment (sgarg-CS) [inline src/test/java/io/cdap/plugin/zendesk/source/batch/ZendeskBatchSourceConfigTest.java]: Add a failure assertion if `ValidationException` is not thrown.

File: src/test/java/io/cdap/plugin/zendesk/source/batch/ZendeskBatchSourceConfigTest.java [hunk 3]
```diff
@@ -322,4 +334,125 @@ public void testGetSchemaInvalidJson() {
...
+      MockFailureCollector mockFailureCollector = new MockFailureCollector();
+      MockArguments mockArguments = new MockArguments();
+      BatchSourceContext context = Mockito.mock(BatchSourceContext.class);
+      Mockito.when(context.getFailureCollector()).thenReturn(mockFailureCollector);
+      Mockito.when(context.getArguments()).thenReturn(mockArguments);
+      Emitter<StructuredRecord> emitter = new MockEmitter<>();
+      KeyValue<NullWritable, StructuredRecord> input = Mockito.mock(KeyValue.class);
+      zendeskBatchSource.transform(input, emitter);
+      zendeskBatchSource.prepareRun(context);
+    } catch (ValidationException ve) {
```
Comment (sgarg-CS) [inline src/test/java/io/cdap/plugin/zendesk/source/batch/ZendeskBatchSourceConfigTest.java:392]: Same comment as above

File: src/test/java/io/cdap/plugin/zendesk/source/batch/http/ZendeskInputFormatProviderTest.java [hunk 4]
```diff
@@ -0,0 +1,78 @@
...
+import io.cdap.plugin.zendesk.source.batch.ZendeskBatchSourceConfig;
+import io.cdap.plugin.zendesk.source.batch.ZendeskInputFormatProvider;
+import org.junit.Assert;
+import org.junit.Test;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Map;
+
+public class ZendeskInputFormatProviderTest {
```
Comment (sgarg-CS) [inline src/test/java/io/cdap/plugin/zendesk/source/batch/http/ZendeskInputFormatProviderTest.java]: Zendesk-related test files can be moved outside `http` package

File: src/test/java/io/cdap/plugin/zendesk/source/batch/http/ZendeskSplitTest.java [hunk 5]
```diff
@@ -0,0 +1,89 @@
...
+    ZendeskSplit zendeskSplit = new ZendeskSplit("domain", "object");
+    Assert.assertEquals("domain", zendeskSplit.getSubdomain());
+    Assert.assertEquals("object", zendeskSplit.getObject());
+  }
+
+  @Test
+  public void testReadFields() throws IOException {
+    ZendeskSplit zendeskSplit = new ZendeskSplit("domain", "object");
+    ObjectInputStream objectInputStream = Mockito.mock(ObjectInputStream.class);
+    ZendeskSplit splits = new ZendeskSplit();
```
Comment (sgarg-CS) [inline src/test/java/io/cdap/plugin/zendesk/source/batch/http/ZendeskSplitTest.java]: `splits` unused variable?

File: src/test/java/io/cdap/plugin/zendesk/source/batch/ZendeskBatchSourceConfigTest.java (hunk 1 above)
Comment (Shubhangi-cs) [reply src/test/java/io/cdap/plugin/zendesk/source/batch/ZendeskBatchSourceConfigTest.java]: Removed

File: src/test/java/io/cdap/plugin/zendesk/source/batch/ZendeskBatchSourceConfigTest.java (hunk 2 above)
Comment (Shubhangi-cs) [reply src/test/java/io/cdap/plugin/zendesk/source/batch/ZendeskBatchSourceConfigTest.java]: Done

File: src/test/java/io/cdap/plugin/zendesk/source/batch/ZendeskBatchSourceConfigTest.java (hunk 3 above)
Comment (Shubhangi-cs) [reply src/test/java/io/cdap/plugin/zendesk/source/batch/ZendeskBatchSourceConfigTest.java:392]: Done

File: src/test/java/io/cdap/plugin/zendesk/source/batch/http/ZendeskInputFormatProviderTest.java (hunk 4 above)
Comment (Shubhangi-cs) [reply src/test/java/io/cdap/plugin/zendesk/source/batch/http/ZendeskInputFormatProviderTest.java]: Moved the files outside the package.

File: src/test/java/io/cdap/plugin/zendesk/source/batch/http/ZendeskSplitTest.java (hunk 5 above)
Comment (Shubhangi-cs) [reply src/test/java/io/cdap/plugin/zendesk/source/batch/http/ZendeskSplitTest.java]: Removed and added a separate test case.

**Pull request merged**